server/src/main/java/org/elasticsearch/health/HealthIndicator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public record HealthIndicator(String name, String component, HealthStatus status, String summary, ToXContentFragment details)
```

Comment:
I would prefer to introduce an explicit `HealthIndicatorDetails` interface, I think it will eventually need to also be writeable and it also just makes it easier to navigate the code.

Reply:
I will implement this as a followup if you do not mind